### PR TITLE
Update example with zio-config

### PR DESCRIPTION
**Pull request by mvelimir**

### CODE REVIEW

File: opentelemetry-example/src/main/scala/zio/telemetry/opentelemetry/example/BackendServer.scala [hunk 1]
```diff
@@ -1,39 +1,50 @@
...
+import zio.telemetry.opentelemetry.example.config.AppConfig
+import zio.telemetry.opentelemetry.example.http.{AppEnv, AppTask, Client, StatusService}
+import zio.{ExitCode, Managed, ZIO, ZLayer}
 import org.http4s.syntax.kleisli._
 import sttp.client.asynchttpclient.zio.AsyncHttpClientZioBackend
+import sttp.model.Uri
 
 object BackendServer extends zio.App {
+  implicit val sttpUriDescriptor: zio.config.magnolia.Descriptor[Uri] =
+    zio.config.magnolia.Descriptor[String].transformOrFailLeft(str =>
```
Comment (mijicd) [inline opentelemetry-example/src/main/scala/zio/telemetry/opentelemetry/example/BackendServer.scala]: I'd import it, e.g.:

```scala
import zio.config.magnolia.Descriptor
```

Same with zio.App.

File: opentelemetry-example/src/main/scala/zio/telemetry/opentelemetry/example/config/ProxyConfig.scala [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package zio.telemetry.opentelemetry.example.config
+
+import sttp.model.Uri
+
+case class ProxyConfig(host: Uri)
```
Comment (mijicd) [inline opentelemetry-example/src/main/scala/zio/telemetry/opentelemetry/example/config/ProxyConfig.scala]: Case classes should always be final.

File: opentelemetry-example/src/main/scala/zio/telemetry/opentelemetry/example/config/TracerHost.scala [hunk 3]
```diff
@@ -0,0 +1,3 @@
+package zio.telemetry.opentelemetry.example.config
+
+case class TracerHost(host: String) extends AnyVal
```
Comment (mijicd) [inline opentelemetry-example/src/main/scala/zio/telemetry/opentelemetry/example/config/TracerHost.scala]: Same here.

**Review by mijicd — CHANGES_REQUESTED** [review]
A few tweaks and it's ready to go.

File: opentelemetry-example/src/main/scala/zio/telemetry/opentelemetry/example/BackendServer.scala (hunk 1 above)
Comment (mvelimir) [reply opentelemetry-example/src/main/scala/zio/telemetry/opentelemetry/example/BackendServer.scala]: Using `Descriptor` directly is deprecated.

Comment (mijicd) [reply opentelemetry-example/src/main/scala/zio/telemetry/opentelemetry/example/BackendServer.scala]: What does that mean? You're using it anyhow, I'm suggesting you to import it, as illustrated [here](https://zio.github.io/zio-config/docs/automatic/automatic_index#is-that-the-only-way-for-custom-derivation-)

Comment (mvelimir) [reply opentelemetry-example/src/main/scala/zio/telemetry/opentelemetry/example/BackendServer.scala]: https://github.com/zio/zio-config/blob/68e41920ad7c3f74af379ebc5dbd228f05ecc43c/magnolia/shared/src/main/scala/zio/config/magnolia/DeriveConfigDescriptor.scala#L661

Comment (mijicd) [reply opentelemetry-example/src/main/scala/zio/telemetry/opentelemetry/example/BackendServer.scala]: Please read what I wrote [here](https://github.com/mvelimir/zio-telemetry/pull/4#discussion_r680906406).

Comment (mvelimir) [reply opentelemetry-example/src/main/scala/zio/telemetry/opentelemetry/example/BackendServer.scala]: I get it now.

**Review by mijicd — APPROVED** [review]

**Pull request merged**